src/createEventEmitterProxy.test.ts

```diff
@@ -160,4 +179,63 @@ describe('createEventEmitterProxy', () => {
 
     expect(proxy.bar()).toStrictEqual([true, 42]);
   });
+
+  it('only migrates events that were added via the proxy, not all the events', () => {
+    const original = new EventEmitter();
+    const next = new EventEmitter();
+
+    let moveCount = 0;
+    original.on('shouldNotMove', () => (moveCount += 1));
+
+    const proxy = createEventEmitterProxy(original);
+    expect(proxy.eventNames()[0]).toBe('shouldNotMove');
+    proxy.on('shouldMove', () => (moveCount += 1));
+    proxy.setTarget(next);
+    next.emit('shouldMove');
+    expect(moveCount).toBe(1);
+    next.emit('shouldNotMove');
+    expect(moveCount).toBe(1);
+  });
+
+  it('does not migrate events that have been removed via "off"', () => {
```

Comment:
should test `removeListener` as well?

Reply:
`off` is an alias to `removeListener` so I didn't test it, but since we are handling removeListener & off specifically, adding something to make sure theres no regression would be wise.